servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/CompletableAmbTest.java

```diff
@@ -31,103 +31,133 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 
-@RunWith(Parameterized.class)
 public class CompletableAmbTest {
 
-    private final TestCompletable first = new TestCompletable();
-    private final TestCompletable second = new TestCompletable();
-    private final TestCompletableSubscriber subscriber = new TestCompletableSubscriber();
-    private final TestCancellable cancellable = new TestCancellable();
+    private TestCompletable first;
+    private TestCompletable second;
+    private TestCompletableSubscriber subscriber;
+    private TestCancellable cancellable;
 
-    public CompletableAmbTest(final BiFunction<Completable, Completable, Completable> ambSupplier) {
+    @BeforeEach
+    void beforeEach() {
+        first = new TestCompletable();
+        second = new TestCompletable();
+        subscriber = new TestCompletableSubscriber();
+        cancellable = new TestCancellable();
+    }
+
+    private void setUp(final BiFunction<Completable, Completable, Completable> ambSupplier) {
         toSource(ambSupplier.apply(first, second)).subscribe(subscriber);
         subscriber.awaitSubscription();
         assertThat("First source not subscribed.", first.isSubscribed(), is(true));
         assertThat("Second source not subscribed.", second.isSubscribed(), is(true));
     }
 
-    @Parameterized.Parameters
     public static Collection<BiFunction<Completable, Completable, Completable>> data() {
         return asList(Completable::ambWith,
                 (first, second) -> amb(first, second),
                 (first, second) -> amb(asList(first, second)));
     }
 
-    @Test
-    public void successFirst() {
+    @ParameterizedTest(name = "{displayName} [{index}]")
+    @MethodSource("data")
+    public void successFirst(final BiFunction<Completable, Completable, Completable> ambSupplier) {
```

Comment:
Enum trick works nicely with displaying parameter. I do prefer customized `name` for `@ParameterizedTest` as it shows method name :

<img width="270" alt="Screen Shot 2021-03-14 at 11 17 07 AM" src="https://user-images.githubusercontent.com/3769695/111079476-68f99200-84b7-11eb-9d11-b73af9cd7aa0.png">

vs default `name`:

<img width="522" alt="Screen Shot 2021-03-14 at 11 16 14 AM" src="https://user-images.githubusercontent.com/3769695/111079504-7d3d8f00-84b7-11eb-97c7-72001b62f575.png">

Reply:
Interesting, I suggested to remove the `{displayName}` because intellij idea shows it in this way for me (changed only `failFirst`):
<img width="632" alt="Screenshot 2021-03-16 at 15 34 40" src="https://user-images.githubusercontent.com/3968288/111389469-42318c00-866e-11eb-899f-c07341e875b2.png">
Because each set of params is grouped under the test name, the `{displayName}` looked redundant.
Works for me either way.

Reply:
This is how junit test report looks like
<img width="1223" alt="Screenshot 2021-03-16 at 15 47 05" src="https://user-images.githubusercontent.com/3968288/111389888-ffbc7f00-866e-11eb-912a-105f25430dae.png">